### PR TITLE
[License] move cma_api to 3rdparty. separate BSD 2-clause and 3-clause

### PR DESCRIPTION
**Pull request by yzhliu**
@tqchen @tmoreau89

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -214,18 +214,22 @@ Apache Software Foundation License 2.0
 3rdparty/dmlc-core
 
 
-BSD License
+BSD 2-clause License
 -----------
```
Comment (tqchen) [inline LICENSE]: make sure the underscore covers all the above line

File: LICENSE [hunk 2]
```diff
@@ -214,18 +214,22 @@ Apache Software Foundation License 2.0
...
 
 3rdparty/picojson
+
+
+BSD 3-clause License
+-----------
+
 3rdparty/dmlc-core/include/dmlc/concurrentqueue.h
```
Comment (tqchen) [inline LICENSE]: Add 3rdparty/dmlc-core/include/dmlc/blockingconcurrentqueue.h
c.f. https://github.com/apache/incubator-mxnet/blob/master/LICENSE#L337

Comment (tqchen) [reply LICENSE]: NOTE: we need to add a license file for blockingconcurrentqueue.h, as part of the file is under zlib, and others under BSD-2

Comment (tqchen) [reply LICENSE]: We should have a section 

BSD 2-clause license + zlib license

File: LICENSE [hunk 3]
```diff
@@ -214,18 +214,22 @@ Apache Software Foundation License 2.0
...
 
 
-BSD License
+BSD 2-clause License
 -----------
 
 3rdparty/picojson
+
+
+BSD 3-clause License
```
Comment (tqchen) [inline LICENSE]: Sorry I double checked, and find that the concurrentqueue.h is under BSD-2 (instead of 3)

**Comment by yzhliu** [comment]
@tqchen please double check.

**Comment by tqchen** [comment]
Thanks @yzhliu I will send a followup PR to refine the blockingconcurrentqueue license

**Pull request merged**